src/components/GlStyleService.js

```diff
@@ -34,49 +28,31 @@ goog.provide('ga_glstyle_service');
 
         this.get = function(styleUrl) {
           var that = this;
-          var defer = $q.defer();
-          $http.get(styleUrl, { cache: true }).then(function(response) {
+          return $http.get(styleUrl, {
+            cache: true
+          }).then(function(response) {
             that.styleCache_ = response.data;
-            if (!that.styleCache_.sprite) {
-              defer.resolve({
-                style: that.styleCache_
-              });
-              return;
-            }
-            var spriteUrl = that.styleCache_.sprite + '.json';
-            that.getSprite(spriteUrl).then(function(styleData) {
-              defer.resolve(styleData);
-            });
+            return response.data;
           }, function(res) {
-            that.styleCache_ = null;
-            that.spriteCache_ = null;
             $window.console.error('Unable to load the style from ' + styleUrl +
                 ' response status is ' + res.status);
-            defer.reject(res);
           });
-          return defer.promise;
         };
 
-        this.getSprite = function(spriteUrl) {
-          var that = this;
+        this.getSpriteData = function(spriteUrl) {
           return $http.get(spriteUrl, { cache: true }).then(function(res) {
-            that.spriteCache_ = res.data;
-            return {
-              style: that.styleCache_,
-              sprite: that.spriteCache_
-            };
+            return res.data;
```

Comment:
why not but, why remove the spriteCache?

Reply:
because:
 - we never use it. 
 - it 's already cache by the $http service.
 - add a complexity in gaGlStyle. 

gaGlStyle returns a raw glStyle object, it's more simple and clear.

Reply:
but i can leave with it if you prefer

Reply:
the sprite is cached by $http, so you  make request only once. and you always get the style you then the sprite. It was the case before this too.